src/ui/components/YfmWrapper/YfmWrapper.tsx

```diff
@@ -1,16 +1,25 @@
 import React from 'react';
 
 import {useLatex} from '@diplodoc/latex-extension/react';
+import {useMermaid} from '@diplodoc/mermaid-extension/react';
 import debounce from 'lodash/debounce';
-import {YFM_LATEX_CLASSNAME} from 'ui/constants';
+import {YfmMetaScripts} from 'shared/constants/yfm';
+import {YFM_LATEX_CLASSNAME, YFM_MERMAID_CLASSNAME} from 'ui/constants';
 import {registry} from 'ui/registry';
 
 import type {YfmWrapperProps} from '../../registry/units/common/types/components/YfmWrapper';
 
+let hasLatexImported = false;
+let hasMermaidImported = false;
+
 export const YfmWrapper = React.forwardRef<HTMLDivElement, Omit<YfmWrapperProps, 'ref'>>(
     (props, forwardedRef) => {
         const YfmWrapperContent = registry.common.components.get('YfmWrapperContent');
         const elementRef = React.useRef<HTMLDivElement | null>(null);
+        const hasLatexScript =
```

Comment:
Maybe move it into useEffect deps and find if one of them is presented only when metaScripts prop changed?